src/client/Client.cc

```diff
@@ -8558,8 +8559,15 @@ loff_t Client::_lseek(Fh *f, loff_t offset, int whence)
     ceph_abort();
   }
 
+  if (f->pos < 0){
+    f->pos = old_pos;
+    r = -EINVAL;
+  }else{
+    r = f->pos;
+  }
+
   ldout(cct, 3) << "_lseek(" << f << ", " << offset << ", " << whence << ") = " << f->pos << dendl;
-  return f->pos;
+  return r;
```

Comment:
That's not a correct way to handle it. _lseek should cap the offset to range 0..size and return capped offset.

Reply:
As I understand it, the input offset  can be negative.
The return value -1 is indicating the error, so do you mean that the return value can only be -1 or a valid positive value?
Thanks

Reply:
In that particular case, there's nothing wrong in seeking file by negative offset, past file beginning. In that case, it should just rewind to 0 and not signal an error condition.

Reply:
Thanks for the advice. I'll update the code and pull request again。

Reply:
You can reopen this PR and just amend your commit.